app/src/main/java/org/mozilla/fenix/search/telemetry/Utils.kt

```diff
@@ -18,10 +18,30 @@ internal fun getTrackKey(
     cookies: List<JSONObject>
 ): String {
     val paramSet = uri.queryParameterNames
-    var code: String? = null
+    var code: String?
+    var codeWithPrefix: String? = null
 
     if (provider.codeParam.isNotEmpty()) {
         code = uri.getQueryParameter(provider.codeParam)
+        if (code.isNullOrEmpty() &&
+            provider.name == "baidu" &&
+            uri.toString().contains("from=")) {
+            code = uri.toString().substringAfter("from=", "")
+                    .substringBefore("/", "")
+        }
+
+        // Telemetry doest not allow code starting with a figure
+        codeWithPrefix = if (code != null && code.isNotEmpty()) {
+            val codeStart = code.first()
+            val figures = listOf('0', '1', '2', '3', '4', '5', '6', '7', '8', '9')
+            if (codeStart in figures) {
```

Comment:
I think you can do `codeStart.isDigit()` and remove `figures` https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/is-digit.html